agent/config.go

```diff
@@ -2127,7 +2127,7 @@ func ReadConfigPaths(paths []string) (*Config, error) {
 // and performs go-sockaddr/template Parse on each known address in case the
 // user specified a template config for any of their values.
 func (c *Config) ResolveTmplAddrs() (err error) {
-	parse := func(addr *string, name string) {
+	parse := func(addr *string, validateIP bool, name string) {
```

Comment:
okay. I can check if config.socketPAth !=nil which makes sure it starts with unix://. Do we also want to validate if the path is a real path? that would break a lot of existing tests, i would have to rewrite them to generate a temp path to pass to the config structs.